dedupe/labeler.py

```diff
@@ -26,155 +26,107 @@
 logger = logging.getLogger(__name__)
 
 
-class Learner(ABC):
-    candidates: TrainingExamples
+class HasCandidates:
+    """Has a read-only list of pairs that we could ask the user to label."""
```

Comment:
candidates is not really read only since we remove records from it.

Reply:
Yeah RO is a bit misleading, it is RO in the sense that you can't modify it directly, but you can modify it via remove(). But just better to leave out the RO part, I think you're right.